config/crds/hive.openshift.io_machinepools.yaml

```diff
@@ -117,8 +117,8 @@ spec:
                       type: object
                     subnets:
                       description: Subnets is the list of subnets to which to attach
```

Comment:
additionally we should enforce that there is also only one public subnet per private subnet 
- this makes sure there is one pair of public+private subnet per AZ.
- no additional public subnets in some AZs that cluster does not already have a private subnet.
- there is also a requirement for the subnet to have some type of tags to make sure kube sees these subnets correctly.

see https://github.com/openshift/enhancements/blob/master/enhancements/installer/aws-customer-provided-subnets.md#subnets

Reply:
Added. Haven't added the check for the expected tag - kube controller does that, however have included a check if subnets cannot be fetched by describe subnets.